website/auth.py

```diff
@@ -543,22 +544,18 @@ def update_profile(user):
 
         # Validations, optional fields
         if 'birth_year' in body:
-            if not isinstance(body.get('birth_year'), int) or body['birth_year'] <= 1900 or body['birth_year'] > datetime.datetime.now().year:
+            try:
+                body['birth_year'] = int(body.get('birth_year'))
```

Comment:
The built-in method int() will try to transform a given value to an int, e.g. int(2.5) becomes 2 and int("1") becomes 1. This means that here we are relaxing the constraint that 'birth_year' is of type int (which was previously there). Is this intentional?

Reply:
We already have a number constraint on the HTML input field. With the new way of handling the data on the front-end all data is send as strings to our back-end, which handles are conversions and checks. So a user shouldn't be able to select a value other than a whole number, if for some reason they do; the back-end catches this.

Reply:
Making the backend validation complementary to the frontend validation is a dangerous endeavor. But the most important thing here is to ensure that you are aware of the details around change you did.

Speaking of this validation. The screenshot indicates that there is a problem with the error message. It is appended instead of injected. I know this is not the main focus of your PR but perhaps you could address that?

Reply:
It's not complementary. The back-end still completely ensures that the data is valid (an integer between 1900 and 2022), however if the user follows the "normal" flow they are unable to enter any value other than an integer. The frontend validation is merely to help the user and reduce server load as users are initially stopped by the browser when entered an invalid value.

I noticed this incorrect error message as well and just committed a fix.